velox/core/PlanNode.h

```diff
@@ -822,15 +822,17 @@ class PartitionedOutputNode : public PlanNode {
       bool replicateNullsAndAny,
       PartitionFunctionFactory partitionFunctionFactory,
       RowTypePtr outputType,
-      PlanNodePtr source)
+      PlanNodePtr source,
+      bool streaming = true)
       : PlanNode(id),
         sources_{{std::move(source)}},
         keys_(keys),
         numPartitions_(numPartitions),
         broadcast_(broadcast),
         replicateNullsAndAny_(replicateNullsAndAny),
         partitionFunctionFactory_(std::move(partitionFunctionFactory)),
-        outputType_(std::move(outputType)) {
+        outputType_(std::move(outputType)),
+        streaming_(streaming) {
```

Comment:
Is "streaming" the right abstraction here? I think one can use POBM with both streaming and materialized exchanges, no?

Similarly, the PR description says "vector vs row based for example", but the format of the exchange data is encapsulated in VectorStreamGroup, not Destination.

Let's sync up offline to review the overall design. It would be nice to also update the PR description to explain the design objectives.